baggage-processor/src/test/java/io/opentelemetry/contrib/baggage/processor/BaggageProcessorCustomizerTest.java

```diff
@@ -113,45 +112,46 @@ private static OpenTelemetrySdk getOpenTelemetrySdk(
                         "none",
                         "otel.logs.exporter",
                         MEMORY_EXPORTER))
-            .addPropertiesSupplier(() -> properties);
-    AutoConfigureUtil.setComponentLoader(
-        sdkBuilder,
-        new ComponentLoader() {
-          @SuppressWarnings("unchecked")
-          @Override
-          public <T> List<T> load(Class<T> spiClass) {
-            if (spiClass == ConfigurableSpanExporterProvider.class) {
-              return Collections.singletonList(
-                  (T)
-                      new ConfigurableSpanExporterProvider() {
-                        @Override
-                        public SpanExporter createExporter(ConfigProperties configProperties) {
-                          return spanExporter;
-                        }
-
-                        @Override
-                        public String getName() {
-                          return MEMORY_EXPORTER;
-                        }
-                      });
-            } else if (spiClass == ConfigurableLogRecordExporterProvider.class) {
-              return Collections.singletonList(
-                  (T)
-                      new ConfigurableLogRecordExporterProvider() {
-                        @Override
-                        public LogRecordExporter createExporter(ConfigProperties configProperties) {
-                          return logRecordExporter;
-                        }
-
-                        @Override
-                        public String getName() {
-                          return MEMORY_EXPORTER;
-                        }
-                      });
-            }
-            return spiHelper.load(spiClass);
-          }
-        });
+            .addPropertiesSupplier(() -> properties)
+            .setComponentLoader(
+                new ComponentLoader() {
+                  @SuppressWarnings("unchecked")
+                  @Override
+                  public <T> List<T> load(Class<T> spiClass) {
+                    if (spiClass == ConfigurableSpanExporterProvider.class) {
+                      return Collections.singletonList(
+                          (T)
```

Comment:
try replacing `(T)` with `spiClass.cast(...)` and see whether this lets you remove the `@SuppressWarnings("unchecked")`